src/commons/controlBar/ControlBarShareButton.tsx

```diff
@@ -36,9 +38,30 @@ export class ControlBarShareButton extends React.PureComponent<ControlBarShareBu
     this.handleChange = this.handleChange.bind(this);
     this.toggleButton = this.toggleButton.bind(this);
     this.shareInputElem = React.createRef();
-    this.state = { keyword: '', isLoading: false };
+    this.state = { keyword: '', isLoading: false, isSuccess: false };
   }
 
+  componentDidMount() {
+    document.addEventListener('keydown', this.handleKeyDown);
```

Comment:
Why are you adding an event listener to the whole DOM?

Reply:
This is the hotkey function of create link, so when pressing  ctrl+shift playground will automatically generate a link and copy to clipboard. I can delete this functionality if it is redundant.

Reply:
I'm not saying the functionality is redundant, I'm saying why are you implementing it this way (by attaching a listener to the whole webpage)?

Reply:
Since we are using React, there are ways to do it without using the JS document API

Reply:
> ctrl+shift

Can I clarify what this hotkey is supposed to be?

Reply:
> Since we are using React, there are ways to do it without using the JS document API

sorry I'm not so familiar with this part. What way do you suggest to implement this? Thanks!

Reply:
> > ctrl+shift
> 
> Can I clarify what this hotkey is supposed to be?

This seems cause no change on current Source Academy website.

Reply:
Agree with @RichDom2185 . We can do this more systematically through a library. See https://github.com/source-academy/frontend/issues/2935 and https://github.com/source-academy/frontend/pull/2937